Tasks/Common/MSBuildHelpers/PathFunctions.ps1

```diff
@@ -168,165 +168,26 @@ function Get-VisualStudio_15_0 {
 
     Trace-VstsEnteringInvocation $MyInvocation
     try {
-        # Short-circuit if the setup configuration class ID isn't registered.
-        Write-Verbose "Testing for class ID."
-        if (!(Test-Path -LiteralPath 'REGISTRY::HKEY_CLASSES_ROOT\CLSID\{177F0C4A-1CD3-4DE7-A32C-71DBBB9FA36D}')) {
-            return
-        }
-
-        # If the type has already been loaded once, then it is not loaded again.
-        Write-Verbose "Adding Visual Studio setup helpers."
-        Add-Type -Debug:$false -TypeDefinition @'
-namespace MSBuildHelpers.VisualStudio.Setup
-{
-    using System;
-    using System.Collections.Generic;
-    using MSBuildHelpers.VisualStudio.Setup.Com;
-
-    public sealed class Instance
-    {
-        public string Description { get; set; }
-
-        public string DisplayName { get; set; }
-
-        public string Id { get; set; }
-
-        public System.Runtime.InteropServices.ComTypes.FILETIME InstallDate { get; set; }
-
-        public string Name { get; set; }
-
-        public string Path { get; set; }
-
-        public Version Version
-        {
-            get
-            {
-                try
-                {
-                    return new Version(VersionString);
+        # Query for the latest 15.0.* version.
+        #
+        # Note, even though VS 15 Update 1 is sometimes referred to as "15.1", the actual installation
+        # version number is 15.0.26403.7.
+        #
+        # Also note, the capability is registered as VisualStudio_15.0, so the following code should
+        # query for 15.0.* versions only.
+        Write-Verbose "Getting latest Visual Studio 15 setup instance."
+        $output = New-Object System.Text.StringBuilder
+        Invoke-VstsTool -FileName "$PSScriptRoot\vswhere.exe" -Arguments "-version [15.0,15.1) -latest -format json" -RequireExitCodeZero 2>&1 |
+            ForEach-Object {
```

Comment:
I was hit by this in my builds, since my build agents do not have a full VS instance installed, but just the Build Tools in them. I think that the Build Tools installation should be supported. After all, we are not using the "Visual Studio Build" task here, but the MsBuild one, where in the former you *might* want to opt into a particular VS installation.